SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ViewDeltaTest.cpp

```diff
@@ -230,6 +230,67 @@ VIEWDELTA_TEST(GIVEN_entity_and_auth_component_in_view_WHEN_authority_lost_THEN_
 	return true;
 }
 
+VIEWDELTA_TEST(GIVEN_entity_and_component_in_view_WHEN_authority_gained_with_no_component_data_THEN_component_removed_from_view)
```

Comment:
could be worth having the cases like 2 components to 1, 1 component removed but a new one added, etc.- it feels like there's a lot of edge cases to this logic